Add info on sync modes to nodes-and-clients docs (#3885)
ethereum/ethereum-org-website: src/content/developers/docs/nodes-and-clients/index.md

Erigon, formerly known as Turbo‐Geth, is a fork of Go Ethereum oriented toward speed and disk‐space efficiency. Erigon is a completely re-architected implementation of Ethereum, currently written in Go but with implementations in other languages planned. Erigon's goal is to provide a faster, more modular, and more optimized implementation of Ethereum. It can perform a full archive node sync using less than 2TB of disk space, in under 3 days

### Synchronization modes {#sync-modes}

> To follow and verify current data in the network, Ethereum client needs to sync with latest network state. This is done by downloading data from peers, cryptographically verifying their integrity, and building a local blockchain database. 

Reviewer: ```suggestion
To follow and verify current data in the network, the Ethereum client needs to sync with the latest network state. This is done by downloading data from peers, cryptographically verifying their integrity, and building a local blockchain database. 
```

> Synchronization modes represent different approaches to this process with various trade-offs. Clients also vary in their implementation details of sync algorithms. Always refer to the official documentation of your chosen client for specifics on implementation.

Reviewer: ```suggestion
Synchronization modes represent different approaches to this process with various trade-offs. Clients also vary in their implementation of sync algorithms. Always refer to the official documentation of your chosen client for specifics on implementation.
```

#### Overview of strategies

> General overview of synchronization approaches used in mainnet ready clients:

Reviewer: ```suggestion
General overview of synchronization approaches used in Mainnet ready clients:
```

    - Gets only the latest state while relying on trust in developers and consensus mechanism.
    - Client ready to use with current network state in a few minutes.
- Snap sync - Implemented by Geth. Using dynamic snapshots served by peers retrieves all the account and storage data without downloading intermediate trie nodes and then reconstructs the Merkle trie locally. [More on Snap](https://github.com/ethereum/devp2p/blob/master/caps/snap.md)
    - Fastest sync strategy developed by Geth, currently its default
    - Saves a lot of disk usage and network bandwidth without sacrificing security.
- Warp sync – Implemented by OpenEthereum. Nodes regularly generate a consensus-critical state snapshot and any peer can fetch these snapshots over the network, enabling a fast sync from this point. [More on Warp](https://openethereum.github.io/Beginner-Introduction#warping---no-warp)
    - Fastest and default sync mode of OpenEthereum relies on static snapshots served by peers.
    - Similar strategy as snap sync but without certain security benefits.

> - Beam sync – Implemented by Nethermind and Trinity. Works like fast sync but also downloads the state data (witnesses) for the latest blocks, which allows you to query the chain within the first few minutes from starting. [More on Beam](https://medium.com/@jason.carver/intro-to-beam-sync-a0fd168be14a)

Reviewer: I think we should either try to define what a witness is somewhere on this page or remove the reference.

    - Syncs state first and enables you to query RPC in a few minutes.
    - Still in development and not fully relyable, background sync is slowed down and RPC responses might fail.

#### Setup in client

> Clients offer rich configuration options to suite your needs. Pick the one which suits you the most based on its level of security, available data, and cost. Apart from the synchronization algorithm, you can also set pruning of different kinds of old data. Pruning enables deleting outdated data, e.g. removing state trie nodes that are unreachable from recent blocks. 

Reviewer: ```suggestion
Clients offer rich configuration options to suit your needs. Pick the one that suits you best based on the level of security, available data, and cost. Apart from the synchronization algorithm, you can also set pruning of different kinds of old data. Pruning enables deleting outdated data, e.g. removing state trie nodes that are unreachable from recent blocks. 
```
